model/src/main/java/org/cbioportal/genome_nexus/model/ProteinLocation.java

```diff
@@ -0,0 +1,91 @@
+package org.cbioportal.genome_nexus.model;
+
+import java.util.*;
+
+public class ProteinLocation
+{
+    private String transcriptId;
+    private Integer start;
+    private Integer end;
+    private String mutationType;
+
+    public ProteinLocation() {}
+
+    public ProteinLocation(String transcriptId, Integer start, Integer end, String mutationType) {
+        this.transcriptId = transcriptId;
+        this.start = start;
+        this.end = end;
+        this.mutationType = mutationType;
+    }
+
+    public String getTranscriptId() {
+        return transcriptId;
+    }
+
+    public Integer getStart() {
+        return start;
+    }
+
+    public Integer getEnd() {
+        return end;
+    }
+
+    public String getMutationType() {
+        return mutationType;
+    }
+
+    public List<Hotspot> filterHotspot(List<Hotspot> hotspots, ProteinLocation proteinLocation){
```

Comment:
This method, and any business logic in general, should be implemented in a service class. In this case, most likely in the `HotspotFilter` class. We try to keep models simple.